documentation/jetty-documentation/src/main/java/org/eclipse/jetty/docs/programming/migration/ServletToHandlerDocs.java

```diff
@@ -0,0 +1,643 @@
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.docs.programming.migration;
+
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import org.eclipse.jetty.http.HttpCookie;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.Trailers;
+import org.eclipse.jetty.io.Content;
+import org.eclipse.jetty.server.Context;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.server.Session;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.CompletableTask;
+import org.eclipse.jetty.util.Fields;
+import org.eclipse.jetty.util.Promise;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+@SuppressWarnings("unused")
+public class ServletToHandlerDocs
+{
+    @SuppressWarnings("InnerClassMayBeStatic")
+    // tag::request[]
+    public class RequestAPIs extends Handler.Abstract
+    {
+        @Override
+        public boolean handle(Request request, Response response, Callback callback) throws Exception
+        {
+            // Gets the request method.
+            // Replaces:
+            //   - servletRequest.getMethod();
+            String method = request.getMethod();
+
+            // Gets the request protocol name and version.
+            // Replaces:
+            //   - servletRequest.getProtocol();
+            String protocol = request.getConnectionMetaData().getProtocol();
+
+            // Gets the full request URI.
+            // Replaces:
+            //   - servletRequest.getRequestURL();
+            String fullRequestURI = request.getHttpURI().asString();
+
+            // Gets the request context.
+            // Replaces:
+            //   - servletRequest.getServletContext()
+            Context context = request.getContext();
+
+            // Gets the context path.
+            // Replaces:
+            //   - servletRequest.getContextPath()
+            String contextPath = context.getContextPath();
+
+            // Gets the request path.
+            // Replaces:
+            //   - servletRequest.getRequestURI();
+            String requestPath = request.getHttpURI().getPath();
+
+            // Gets the request path after the context path.
+            // Replaces:
+            //   - servletRequest.getServletPath() + servletRequest.getPathInfo()
+            String pathInContext = Request.getPathInContext(request);
```

Comment:
We should document somewhere.... not here... the new policy of not having multiple paths to access the same data, so that wrapping is simpler.   Such a note could be referenced by doco describing these new static methods to explain why they are static.

Reply:
> explain why they are static

I would be interested in this as well.

Reply:
Documented at `pg-server-http-handler-impl-request`.